lib/graphql/schema/union.rb

```diff
@@ -5,31 +5,51 @@ class Union < GraphQL::Schema::Member
       extend GraphQL::Schema::Member::AcceptsDefinition
 
       class << self
-        def possible_types(*types)
+        def possible_types(*types, context: GraphQL::Query::NullContext, **options)
```

Comment:
I find this method confusing now, it seems to be doing two very different things in the two if/else branches?

Reply:
Isn't it doing the same thing at a high-level? The two branches are:

1. setting the possible types
2. getting the possible types

That's still happening but with more logic now:

1. setting the possible types via type memberships
2. getting the possible types taking into account type membership visiblity

I do think it's overloaded but it's still consistent with how it always worked I'd argue?

Reply:
🤔 👍